theano/tensor/nnet/nnet.py

```diff
@@ -302,39 +301,28 @@ class SoftmaxGrad(gof.Op):
     def make_node(self, dy, sm):
         dy = tensor.as_tensor_variable(dy)
         sm = tensor.as_tensor_variable(sm)
-        if dy.type.ndim not in (1, 2) \
-                or dy.type.dtype not in tensor.float_dtypes:
-            raise ValueError('dy must be 1-d or 2-d tensor of floats. Got ',
-                             dy.type)
-        if dy.ndim == 1:
-            dy = tensor.shape_padleft(dy, n_ones=1)
-        if sm.ndim == 1:
-            sm = tensor.shape_padleft(sm, n_ones=1)
```

Comment:
Some tests would assume that the softmax of a vector is a row, so that `softmax(v)[arange(1), y]` still inserts the `cross_entropy_softmax...` Op.
It is OK to change it in this PR, but we will have to handle the transition.